testing/testcafe/helpers/screenshort-comparer.ts

```diff
@@ -0,0 +1,256 @@
+  etalonFileName: string;
+  maskFileName: string;
+  options: ComparerOptions; }) {
+  let equal = false;
+  let attempt = 0;
+  let screenshotBuffer;
+  while (!equal && attempt < options.attempts) {
+    attempt += 1;
+    if (attempt > 1) {
+      fs.unlinkSync(screenshotFileName);
+    }
+    await (element
+      ? t.takeElementScreenshot(element, screenshotFileName)
+      : t.takeScreenshot(screenshotFileName));
+    screenshotBuffer = await getMaskedScreenshotBuffer({
+      screenshotFileName, etalonFileName, maskFileName,
+    });
+    equal = await looksSame({
+      etalonFileName,
+      screenshotBuffer,
+      comparisonOptions: options.looksSameComparisonOptions,
+    });
+    if (attempt < options.attempts) {
+      await t.wait(options.attemptTimeout);
+    }
+  }
+  return { equal, screenshotBuffer };
+}
+
+export async function compareScreenshot(
+  t: TestController,
+  screenshotName: string,
+  element: SelectorType = null,
+  comparisonOptions?: Partial<ComparerOptions>,
+) {
+  const screenshotFileName = path.join(screenshotsPath, screenshotName);
+  const etalonsPath = path.join(path.dirname((t as any).testRun.test.testFile.filename), 'etalons');
+  const etalonFileName = path.join(etalonsPath, screenshotName);
+  const maskFileName = path.join(etalonsPath, screenshotName.replace('.png', '_mask.png'));
+  const options = {
+    ...screenshotComparerDefault,
+    ...(comparisonOptions || {}),
+  } as typeof screenshotComparerDefault;
+  try {
+    ensureArtifactsPath();
+    const { equal, screenshotBuffer } = await tryGetValidScreenshot({
+      t, element, screenshotFileName, etalonFileName, maskFileName, options,
+    });
+    if (!equal) {
+      const diffFileName = path.join(artifactsPath, screenshotName.replace('.png', '_diff.png'));
+      const diffMaskFileName = path.join(artifactsPath, screenshotName.replace('.png', '_mask.png'));
+      const diffBuffer = await getDiff({
+        etalonFileName, screenshotBuffer, options,
+      });
+      const maskBuffer = getMask(diffBuffer, options);
+      fs.writeFileSync(diffFileName, diffBuffer);
+      fs.writeFileSync(diffMaskFileName, maskBuffer);
+      saveArtifacts({ screenshotFileName, etalonFileName });
+      return false;
+    }
+    return true;
+  } catch (e) {
+    saveArtifacts({ screenshotFileName, etalonFileName });
+    throw e;
+  }
+}
+export function createScreenshotsComparer(t: TestController) {
+  const errorMessages: string[] = [];
+  return {
+    takeScreenshot: async (screenshotName: string,
+      element: SelectorType = null,
+      comparisonOptions?: Partial<ComparerOptions>,
+    ) => {
+      try {
+        const isValid = await compareScreenshot(t, screenshotName, element, comparisonOptions);
+        if (!isValid) {
+          errorMessages.push(`Screenshort:'${screenshotName}' isn't valid`);
+        }
+      } catch (e) {
+        errorMessages.push(`Screenshort:'${screenshotName}' isn't valid, internalError: ${e.message}`);
```

Comment:
I'd say "Screenshort:'${screenshotName}' validation failed, internal error: ${e.message}"
Also pay attention at the typo: Screensho>r<t